feature/amazon/src/test/java/com/revenuecat/purchases/amazon/AmazonCacheTest.kt

```diff
@@ -0,0 +1,157 @@
+                JSONObject(it)
+            }
+        }
+
+        override fun putString(
+            cacheKey: String,
+            value: String
+        ) {
+            stubCache[cacheKey] = value
+        }
+    }
+
+    @Before
+    fun setup() {
+        cache = MockDeviceCache(mockk(), apiKey)
+        underTest = AmazonCache(cache)
+    }
+
+    @Test
+    fun `getting cached term skus when there is nothing cached`() {
+        val receiptTermSkus = underTest.getReceiptSkus()
+
+        assertThat(receiptTermSkus).isEmpty()
+    }
+
+    @Test
+    fun `getting cached term skus when there is termskus cached`() {
+        val expected = mapOf(
+            "1234abcdreceiptid" to "com.revenuecat.subscription.weekly",
+            "4321abcdreceiptid" to "com.revenuecat.subscription.monthly"
+        )
+
+        val cachedReceiptsToTermSkusJSON = getStoredJSONFromMap(expected)
+
+        cache.stubCache[underTest.amazonPostedTokensKey] = cachedReceiptsToTermSkusJSON
+
+        val receiptTermSkus = underTest.getReceiptSkus()
+
+        assertThat(receiptTermSkus).isEqualTo(
+            mapOf(
+                "1234abcdreceiptid" to "com.revenuecat.subscription.weekly",
+                "4321abcdreceiptid" to "com.revenuecat.subscription.monthly"
+            )
+        )
+    }
+
+    @Test
+    fun `set receipt term skus on an empty cache`() {
+        val expected = mapOf(
+            "1234abcdreceiptid" to "com.revenuecat.subscription.weekly",
+            "4321abcdreceiptid" to "com.revenuecat.subscription.monthly"
+        )
+
+        underTest.setReceiptSkus(expected)
+
+        val actualStoredJSON = JSONObject(cache.stubCache[underTest.amazonPostedTokensKey])
+        val actualStoredMapAsJSON = actualStoredJSON["receiptsToSkus"] as JSONObject
+
+        assertThat(actualStoredMapAsJSON).isNotNull
+        assertThat(actualStoredMapAsJSON.keys().asSequence().count()).isEqualTo(expected.size)
+
+        expected.forEach { (key, value) ->
+            assertThat(actualStoredMapAsJSON[key]).isEqualTo(value)
+        }
+    }
+
+    @Test
+    fun `set receipt term skus on a non empty cache`() {
+        val alreadyCached = mapOf(
+            "1234abcdreceiptid" to "com.revenuecat.subscription.weekly"
+        )
+
+        val cachedReceiptsToTermSkusJSON = getStoredJSONFromMap(alreadyCached)
+        cache.stubCache[underTest.amazonPostedTokensKey] = cachedReceiptsToTermSkusJSON
+
+        val newToCache = mapOf(
+            "4321abcdreceiptid" to "com.revenuecat.subscription.monthly"
+        )
+
+        underTest.setReceiptSkus(newToCache)
+
+        val actualStoredJSON = JSONObject(cache.stubCache[underTest.amazonPostedTokensKey])
+        val actualStoredMapAsJSON = actualStoredJSON["receiptsToSkus"] as JSONObject
+
+        assertThat(actualStoredMapAsJSON).isNotNull
+
+        val expected = alreadyCached + newToCache
+
+        assertThat(actualStoredMapAsJSON.keys().asSequence().count()).isEqualTo(expected.size)
```

Comment:
would just getting the [length](https://stleary.github.io/JSON-java/org/json/JSONObject.html#length--) work as a shortcut? 

i.e.: 
```kotlin
assertThat(actualStoredMapAsJSON.length()).isEqualTo(expected.size)
```